common/ast/src/ast/statement.rs

```diff
@@ -206,6 +201,15 @@ pub enum InsertSource<'a> {
     Select { query: Box<Query<'a>> },
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct CreateDatabaseStmt<'a> {
+    pub if_not_exists: bool,
+    pub catalog: Option<Identifier<'a>>,
+    pub database: Identifier<'a>,
+    pub engine: DatabaseEngine,
```

Comment:
Is it better to make the engine optional (so that the ast will display as origin)? If so, we can remove the Default in DatabaseEngine.